### PR TITLE
(#6355) - extract changes filter into its own plugin

### PR DESCRIPTION
**Pull request by nolanlawson**
All functionality related to changes or replication filtering, such as `view`, `filter`, `selector`, etc. has been moved into its own plugin, `pouchdb-changes-filter`.

This PR is a non-breaking change. The idea is to make it trivial in PouchDB 7.0.0 to remove this new plugin (`pouchdb-changes-filter`) from `pouchdb-core`.

To give an idea of the kind of size savings we may see in PouchDB 7.0, I ran some stats. This assumes that in 7.0 we will remove `pouchdb-mapreduce`, `pouchdb-debug`, and `pouchdb-changes-filter` from core:

<table border="0" cellspacing="0" cellpadding="0" class="ta1"><colgroup><col width="320"/><col width="99"/><col width="99"/><col width="99"/><col width="99"/></colgroup><tr class="ro1"><td style="text-align:left;width:207.84pt; " class="ce1"><p>Packages</p></td><td style="text-align:left;width:64.01pt; " class="ce1"><p>Size (min)</p></td><td style="text-align:left;width:64.01pt; " class="ce1"><p>Size (min+gz)</p></td><td style="text-align:left;width:64.01pt; " class="Default"><p>Diff (min)</p></td><td style="text-align:left;width:64.01pt; " class="Default"><p>Diff (min+gz)</p></td></tr><tr class="ro2"><td style="text-align:left;width:207.84pt; " class="ce2"><p>pouchdb 6.1.2 (old baseline, using master)</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>147420</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>47211</p></td><td style="text-align:left;width:64.01pt; " class="Default"> </td><td style="text-align:left;width:64.01pt; " class="Default"> </td></tr><tr class="ro2"><td style="text-align:left;width:207.84pt; " class="ce2"><p>pouchdb 7.0 (new baseline, using this PR)</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>120314</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>38291</p></td><td style="text-align:left;width:64.01pt; " class="Default"> </td><td style="text-align:left;width:64.01pt; " class="Default"> </td></tr><tr class="ro2"><td style="text-align:left;width:207.84pt; " class="ce2"><p>7.0 baseline +pouchdb-mapreduce</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>135274</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>43397</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>14960</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>5106</p></td></tr><tr class="ro1"><td style="text-align:left;width:207.84pt; " class="ce2"><p>7.0 baseline +pouchdb-debug</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>126399</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>40497</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>6085</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>2206</p></td></tr><tr class="ro1"><td style="text-align:left;width:207.84pt; " class="ce2"><p>7.0 baseline +pouchdb-changes-filter</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>127302</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>40508</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>6988</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>2217</p></td></tr><tr class="ro3"><td style="text-align:left;width:207.84pt; " class="ce2"><p>7.0 baseline +pouchdb-debug,pouchdb-changes-filter,pouchdb-mapreduce</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>148072</p></td><td style="text-align:right; width:64.01pt; " class="ce2"><p>47508</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>27758</p></td><td style="text-align:right; width:64.01pt; " class="Default"><p>9217</p></td></tr></table>

Also I can confirm that the resulting 7.0 `pouchdb.js` package contains no `eval()`, `new Function()`, etc., which will make extension developers very happy. 😃 Those only exist in `pouchdb-mapreduce` and `pouchdb-changes-filter` now.

### CODE REVIEW

File: packages/node_modules/pouchdb-ajax/tmp.js [hunk 1]
```diff
@@ -1 +0,0 @@
-testUtils.ajax = require('./lib/index-browser')
```
Comment (garrensmith) [inline packages/node_modules/pouchdb-ajax/tmp.js]: Is this file needed?

Comment (nolanlawson) [inline packages/node_modules/pouchdb-ajax/tmp.js]: no looks like this was checked in by accident

**Comment by nolanlawson** [comment]
Had to re-rebase, waiting for a green to be extra sure.
